lib/taurus/core/util/log.py

```diff
@@ -793,6 +793,15 @@ def _format_stack(self, stack_func=inspect.stack):
                         out += '<could not find suitable string representation>'
         return out
 
+    def loggingtofile(self, msg, logging_path, *args, **kw):
+
+        file_name = logging_path + "/spock_session.log"
+
+        logging_file = open(file_name, 'a')
+        logging_file.write('\n')
+        logging_file.write(msg)
+        logging_file.close()
+
```

Comment:
Without knowing the uses cases in Sardana. As @cpascual  pointed, the logging module allows you to redirect the output to a file or add a new (file) handler. See [1], and [2]

So, HIMO, I would not write the log with taurus.

Take in mind that Taurus.Logger provides the `getLogger (name)` method to get the appropriate Logger (e.g to access a subLogger),  so you can implement a solution based on  a `file handler` using the `logging` API (addHandler, removeHandler).

I will check the `log_to_file` uses and I'm coming with more ideas.

[1] https://docs.python.org/2/howto/logging.html#logging-to-a-file
[2] https://stackoverflow.com/questions/13733552/logger-configuration-to-log-to-file-and-print-to-stdout